crates/wit-parser/src/sizealign.rs

```diff
@@ -1,26 +1,210 @@
+use std::{
+    num::NonZeroUsize,
+    ops::{Add, AddAssign},
+};
+
 use crate::{FlagsRepr, Int, Resolve, Type, TypeDef, TypeDefKind};
 
-#[derive(Default)]
-pub enum AddressSize {
-    #[default]
-    Wasm32,
-    Wasm64,
+/// Architecture specific alignment
+#[derive(Eq, PartialEq, PartialOrd, Clone, Copy, Debug)]
+pub enum Alignment {
+    Pointer,
+    Bytes(NonZeroUsize),
 }
```

Comment:
Mind expanding the documentation here explaining the representation? Basically saying that this is intended to be a pointer-width-agnostic representation of alignment.